src/journal/journald-server.c

```diff
@@ -1378,7 +1378,7 @@ int server_process_datagram(sd_event_source *es, int fd, uint32_t revents, void
 
         if (fd == s->syslog_fd) {
                 if (n > 0 && n_fds == 0)
-                        server_process_syslog_message(s, strstrip(s->buffer), ucred, tv, label, label_len);
+                        server_process_syslog_message(s, s->buffer + strspn(s->buffer, WHITESPACE), ucred, tv, label, label_len);
```

Comment:
So, I much prefer @evverx's suggestion I must say: pass the original string as it is to the syslog forwarder, but use the strstrip()ped version for our own purposes. And I also agree with @evverx suggestion that we should probably also forward any NUL bytes that might be in the string. hence, change  server_process_syslog_message() to take a buffer length parameter in addition to the buffer address, and then pass that on as it is, entirely unmodified to forward_to_syslog_raw() inside of server_process_syslog_message(), and then immediately after strip all the crap away, including trailing and leading whitespace as well as everything after the first NUL byte... It's fine passing something like that on afte rall, but I think it's generally safer to truncate on NUL and not be happy with something like that for our own purposes (in particular as the syslog RFC suggests using the NUL byte as message separator when syslog on TCP is uses).